subsys/net/lib/coap/coap.c

```diff
@@ -1706,8 +1709,9 @@ struct coap_pending *coap_pending_next_to_expire(
 static uint32_t init_ack_timeout(const struct coap_transmission_parameters *params)
 {
 #if defined(CONFIG_COAP_RANDOMIZE_ACK_TIMEOUT)
-	const uint32_t max_ack = params->ack_timeout *
-				 CONFIG_COAP_ACK_RANDOM_PERCENT / 100;
+	const uint16_t random_percent = params->ack_random_percent ? params->ack_random_percent
+								   : CONFIG_COAP_ACK_RANDOM_PERCENT;
+	const uint32_t max_ack = params->ack_timeout * random_percent / 100;
```

Comment:
Please change 100 to 100U so we do not need to change it later because of some tool complaining.
```
const uint32_t max_ack = params->ack_timeout * random_percent / 100U;
```